docs/design/gluster-block-provisioning.md

```diff
@@ -0,0 +1,350 @@
+* name: name given to the volume
+* volume: the id of the block-hosting volume where the loopback file resides
+* hosts: the target ips for this volume
+
+#### Cluster selection
+
+By default, heketi would consider all available clusters
+when looking for space to create a new block-volume file.
+
+With the clusters request parameter, this search can be
+narrowed down to an explicit list of one or more clusters.
+With the help of the ```clusterid``` storage class option,
+this gives the kubernetes administrator a way to e.g. separate
+different storage qualities, or to reserve a cluster
+exclusively for block-volumes.
+
+
+### Details On Calling ```gluster-block```
+
+Heketi calls out to ```gluster-block``` the same way it
+currently calls out to the standard gluster cli for the
+normal volume create operations, i.e. it uses a kubexec
+mechanism to run the command on one of the gluster nodes.
+(In a non-kubernetes install, it uses ssh.)
+
+
+## Details About ```gluster-block```
+
+```gluster-block``` is the gluster-level tool to make creation
+and consumption of block volumes very easy. It consists of
+a server component ```gluster-blockd``` that runs on the gluster
+storage nodes and a command line client utility ```gluster-block```
+that talks to the ```gluster-blockd``` with local RPC mechanism
+and can be invoked on any of the gluster storage nodes.
+
+gluster-block takes care of creating loopback files on the
+specified gluster volume. These volumes are then exported
+as iSCSI targets with the help of the tcmu-runner mechanism,
+using the gluster backend with libgfapi. This has the big
+advantage that it is talking to the gluster volume directly
+in user space without the need of a glusterfs fuse mount,
+skipping the kernel/userspace context switches and the
+user-visible mount altogether.
+
+The supported operations are:
+
+* create
+* list
+* info
+* delete
+* modify
+
+Details about the gluster-block architecture can be found
+in the gluster-block git repository <https://github.com/gluster/gluster-block>
+and the original design notes
+<https://docs.google.com/document/d/1psjLlCxdllq1IcJa3FN3MFLcQfXYf0L5amVkZMXd-D8/edit?usp=sharing>.
+
+
+## Details About Using The In-Tree iSCSI Mount Plugin
+
+The kubernetes in-tree iSCSI mount plugin is used by the PVs created
+by the new glusterblock external provisioner. The tasks that are performed
+when a user brings up an application pod using a corresponding glusterblock
+PVC are:
+
+* create /dev/sdX on the host (iscsi login / initiator)
+* format the device if it is has not been formatted
+* mount the file system on the host
+* bind-mount the host mounted directory into the application pod
+
+Multi-pathing support has been added to the iSCSI plugin so that
+the gluster-block volumes are made highly availble with the ```hacount```
+feature.
+
+The iSCSI plugin has been changed in such a way that it does not allow
+more than one concurrent R/W mount of an iscsi block device.
```

Comment:
Can we add sections on:
- how we'll be securing the RWO volumes?
- how we'll be scaling?